packages/react-instantsearch-hooks-server/src/__tests__/getServerState.test.tsx

```diff
@@ -0,0 +1,728 @@
+          facetFilters: [['brand:Apple']],
+          facets: ['brand'],
+          highlightPostTag: '__/ais-highlight__',
+          highlightPreTag: '__ais-highlight__',
+          maxValuesPerFacet: 10,
+          query: 'iphone',
+          tagFilters: '',
+        },
+      },
+      {
+        indexName: 'instant_search_price_asc',
+        params: {
+          analytics: false,
+          attributesToHighlight: [],
+          attributesToRetrieve: [],
+          attributesToSnippet: [],
+          clickAnalytics: false,
+          facets: 'brand',
+          highlightPostTag: '__/ais-highlight__',
+          highlightPreTag: '__ais-highlight__',
+          hitsPerPage: 1,
+          maxValuesPerFacet: 10,
+          page: 0,
+          query: 'iphone',
+          tagFilters: '',
+        },
+      },
+      {
+        indexName: 'instant_search_rating_desc',
+        params: {
+          facetFilters: [['brand:Apple']],
+          facets: ['brand'],
+          highlightPostTag: '__/ais-highlight__',
+          highlightPreTag: '__ais-highlight__',
+          maxValuesPerFacet: 10,
+          query: 'iphone',
+          tagFilters: '',
+        },
+      },
+      {
+        indexName: 'instant_search_rating_desc',
+        params: {
+          analytics: false,
+          attributesToHighlight: [],
+          attributesToRetrieve: [],
+          attributesToSnippet: [],
+          clickAnalytics: false,
+          facets: 'brand',
+          highlightPostTag: '__/ais-highlight__',
+          highlightPreTag: '__ais-highlight__',
+          hitsPerPage: 1,
+          maxValuesPerFacet: 10,
+          page: 0,
+          query: 'iphone',
+          tagFilters: '',
+        },
+      },
+      {
+        indexName: 'instant_search_price_desc',
+        params: {
+          facetFilters: [['brand:Apple']],
+          facets: ['brand'],
+          highlightPostTag: '__/ais-highlight__',
+          highlightPreTag: '__ais-highlight__',
+          maxValuesPerFacet: 10,
+          query: 'iphone',
+          tagFilters: '',
+        },
+      },
+      {
+        indexName: 'instant_search_price_desc',
+        params: {
+          analytics: false,
+          attributesToHighlight: [],
+          attributesToRetrieve: [],
+          attributesToSnippet: [],
+          clickAnalytics: false,
+          facets: 'brand',
+          highlightPostTag: '__/ais-highlight__',
+          highlightPreTag: '__ais-highlight__',
+          hitsPerPage: 1,
+          maxValuesPerFacet: 10,
+          page: 0,
+          query: 'iphone',
+          tagFilters: '',
+        },
+      },
+    ]);
+  });
+
+  test('returns initialResults', async () => {
+    const searchClient = createSearchClient();
+    const { App } = createTestEnvironment({ searchClient });
+
+    const serverState = await getServerState(<App />);
+
+    expect(Object.keys(serverState.initialResults)).toHaveLength(4);
+    expect(serverState.initialResults).toMatchInlineSnapshot(`
+Object {
```

Comment:
That's typically a good use of snapshot here. We want to assert any regressions on the initial results that are returned, and can't afford to just test some properties.

I put this test last so that the inline snapshot doesn't obstruct other tests. I'm OK with using an external snapshot too if you prefer.